### ISSUE TITLE
DeepChem Docker Image doesn't support CPU training

### ISSUE THREAD

**Issue by rbharath** [deepchem/deepchem#2185]
I was playing around recently with the latest DeepChem docker image on a CPU-only machine. It works as expected, but the image fails when training on the benchmark example with the following error

```
-------------------------------------
Benchmark on dataset: tox21                                                                           
-------------------------------------
Splitting function: random                                                                            
/deepchem/deepchem/data/data_loader.py:162: FutureWarning: featurize() is deprecated and has been renamed to create_dataset().featurize() will be removed in DeepChem 3.0
  "featurize() will be removed in DeepChem 3.0", FutureWarning)                                                                                                                                             
[00:04:04] WARNING: not removing hydrogen atom without neighbors
2020-09-23 00:04:28.455460: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such fi
le or directory; LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64
2020-09-23 00:04:28.455614: E tensorflow/stream_executor/cuda/cuda_driver.cc:313] failed call to cuInit: UNKNOWN ERROR (303)
2020-09-23 00:04:28.455766: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (d4dc7430b2f3): /proc/driver/nvidia/version does not exist
2020-09-23 00:04:28.460013: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-09-23 00:04:28.495332: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 1704000000 Hz
2020-09-23 00:04:28.496068: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x7fad10000b20 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-09-23 00:04:28.496110: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
/miniconda/envs/deepchem/lib/python3.6/site-packages/tensorflow/python/framework/indexed_slices.py:434: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a 
large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
```

I believe this is due to us specifying the GPU install for the latest Dockerfile https://github.com/deepchem/deepchem/blob/master/docker/master/Dockerfile#L21. We should change this so that there's a docker build for CPU-only users. Perhaps we could have two docker releases, one `latest-gpu`, one `latest-cpu` perhaps?

@nd-02110114 I'd love your feedback on whether there's a clean way to do this :)

**Comment by nd-02110114** [comment]
>  It works as expected, but the image fails when training on the benchmark example with the following error

So, you mean the benchmark script itself works well...? 

I think the above message is not a problem. I seem the CPU training works well. From my understanding, TensorFlow and PyTorch automatically switch the CPU training in the case that you install their packages for GPU but the device doesn't have any GPU resources. 

> Perhaps we could have two docker releases, one latest-gpu, one latest-cpu perhaps?

Yes, we could. I think the advantage is image size. The image size for CPU is smaller than for GPU. But, I think there are few users which use the docker images, so I didn't create two images.

**Comment by nd-02110114** [comment]
After reading [this stackoverflow thread](https://stackoverflow.com/questions/47068709/your-cpu-supports-instructions-that-this-tensorflow-binary-was-not-compiled-to-u), I seem the following error indicated the TensorFlow was not optimized for CPU.

```
Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
```

So, the benchmark script itself works well, but the performance may not be good. I think we should provide the both CPU and GPU images.

**Comment by peastman** [comment]
I think that's just the way they compile their standard binaries.  For a long time they didn't even compile for AVX, since they were trying to support very old CPUs.  Eventually they started including that, but still don't include AVX2.  It's not a problem.

**Comment by nd-02110114** [comment]
@rbharath 
I think this issue was resolved. Could we close the issue?
